website/docs/development/release.md

```diff
@@ -6,9 +6,28 @@ sidebar_label: Release process
 
 The release process may be automated in the future.
 
-- Checkout master
-- Update the Hydra version in `hydra/__init__.py`
-- Update NEWS.md with towncrier
-- Create a pip package for hydra-core: `python setup.py sdist bdist_wheel`
+The versioning strategy is that the repo is configured to be on the next version and the version should be bumped AFTER the release, not before.
+
+One should check that the version is the correct one against pypi before releasing.
+
+### Hydra Release Process
+
+- Checkout the release branch.
+- Check the version in `hydra/__init__.py`. This should be one version higher than the version available on PyPI. If it is the same version as PyPI, bump the version locally.
+- Create draft release notes using `towncrier build --draft`.
+- If the draft notes are correct, update NEWS.md using `towncrier build`.
+- Push the news file to release branch.
+- Create a pip package: `python setup.py sdist bdist_wheel`
+- Upload pip package: `python -m twine upload dist/*`
+- Bump the version in `hydra/__init__.py` and push to master.
```

Comment:
You need to update the version in the release branch, not in master.

Reply:
Fixed.